File: backend/src/main/java/ru/nsu/brusn/smpltodo/model/dto/request/CreateNewTaskRequest.java
package ru.nsu.brusn.smpltodo.model.dto.request;

import com.sun.istack.NotNull;
import lombok.Getter;

import java.time.ZonedDateTime;
import java.util.Date;

@Getter
public class CreateNewTaskRequest {
    @NotNull
    private Long folderId;

    @NotNull
    private String name;

    private ZonedDateTime startDate;
    private ZonedDateTime deadline;
    private String description;
}

File: smpltodo/app/src/main/java/ru/nsu/brusn/smpltodo/api/model/dto/request/CreateNewTaskRequest.kt
package ru.nsu.brusn.smpltodo.api.model.dto.request

import com.squareup.moshi.Json

data class CreateNewTaskRequest(
    @field:Json(name = "folderId")
    val folderId: Long,

    @field:Json(name = "name")
    val name: String,
)